service/nodeconfig/framework.go

```diff
@@ -20,26 +20,13 @@ type FrameworkConfig struct {
 	K8sExtClient apiextensionsclient.Interface
 	Logger       micrologger.Logger
 
-	Name string
+	ProjectName string
 }
 
 func NewFramework(config FrameworkConfig) (*framework.Framework, error) {
 	if config.G8sClient == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
 	}
-	if config.K8sClient == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
-	}
-	if config.K8sExtClient == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.K8sExtClient must not be empty", config)
-	}
-	if config.Logger == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
-	}
-
-	if config.Name == "" {
-		return nil, microerror.Maskf(invalidConfigError, "%T.Name must not be empty", config)
-	}
```

Comment:
We should only validate what we use in the current layer. Here we only propagate and the next layer can decide if it needs something and thus has to validate input.

Reply:
Do you think it's maintainable? I think it's very easy to lose track what is validated where. I don't see a problem in validating things multiple times.

Reply:
IMO this is an indicator of us not knowing what we are doing. We should only validate at the end of the chain. Otherwise we validate ourselves to death at some point. I think it is maintainable because here you only have to check if the thing that is validated is actually used or only forwarded. In this case the things I removed the validation checks for are only forwarded, which is why I removed the validation checks.